server/grpc/server_test.go

```diff
@@ -99,6 +101,39 @@ func (s *IntegrationTestSuite) TestGRPCServer() {
 	s.Require().True(servicesMap["cosmos.bank.v1beta1.Query"])
 }
 
+// Test and enforce that we upfront reject any connections to baseapp containing
+// invalid initial x-cosmos-block-height that aren't positive  and in the range [0, max(int64)]
+// See issue https://github.com/cosmos/cosmos-sdk/issues/7662.
+func (s *IntegrationTestSuite) TestGRPCServerInvalidHeaderHeights() {
+	t := s.T()
+	val0 := s.network.Validators[0]
+
+	// We should reject connections with invalid block heights off the bat.
+	invalidHeightStrs := []string{
```

Comment:
can we also add a test for valid heights (latest = 0 and valid heights > 0)?

Reply:
I believe those tests exist in the test suites in this file.